tests/Neo.VM.Tests/UT_ReferenceCounter.cs

```diff
@@ -237,18 +237,18 @@ public void TestArrayNoPush()
             using ExecutionEngine engine = new();
             engine.LoadScript(sb.ToArray());
             Assert.AreEqual(0, engine.ReferenceCounter.Count);
-            Array array = new(engine.ReferenceCounter, new StackItem[] { 1, 2, 3, 4 });
-            Assert.AreEqual(array.Count, engine.ReferenceCounter.Count);
+            Array array = new(new StackItem[] { 1, 2, 3, 4 });
+            engine.CurrentContext.EvaluationStack.Push(array);
+            Assert.AreEqual(array.Count + 1, engine.ReferenceCounter.Count);
```

Comment:
We should try to avoid the UT changes, in this case, what's wrong exactly?

Reply:
this is because of how we process reference counter for compound type, previously we can test the RC without the engine, cause  stackitem itself will process the RC, but in the new version, only executionstack will process the stack, as a result, we must push the item to the stack to get a RC count. 

That is why it was 4 in the past, as the array is not in the stack.
While new version its 5 (4+1), cause we have to push the array itself to the stack.